docs/cddl/diffs/common/votes-certificates.md

```diff
@@ -0,0 +1,137 @@
+# Votes and Certificates - Leios CDDL Changes
+
+Leios introduces a new BLS-based voting system with certificates for endorser block validation.
+
+## Certificate Structure
+
+Leios certificates are embedded in Ranking Blocks as described in [Ranking Blocks - CDDL Changes](ranking-blocks.md). Here is the complete certificate structure:
+
+```cddl
+; Complete Leios certificate structure (from crypto-benchmarks implementation)
+leios_certificate =
+  [ election_id            : election_id                             ; 8-byte election identifier (EID)
+  , endorser_block_hash    : hash32                                  ; Hash of the endorsed block (EB)  
+  , persistent_voters      : [* persistent_voter_id]                 ; Set of persistent voter IDs
+  , nonpersistent_voters   : {* pool_id => bls_signature}            ; Non-persistent voters with eligibility proofs
+  , ? aggregate_elig_sig   : bls_signature                           ; Optional aggregate eligibility signature
+  , aggregate_vote_sig     : bls_signature                           ; Aggregate BLS signature on (election_id || endorser_block_hash)
+  ]
+```
+<sub>[1] [Certificate Reference Implementation](https://github.com/input-output-hk/ouroboros-leios/blob/main/crypto-benchmarks.rs/src/cert.rs#L13-L21), [2] [Certificate Abstract Interface](https://github.com/input-output-hk/ouroboros-leios-formal-spec/blob/main/formal-spec/Leios/Base.agda#L24-L28)</sub>
+
+## Vote Structure
+
+The Leios voting system supports two types of voters: persistent voters (selected per epoch) and non-persistent voters (selected per election via local sortition).
+
+```cddl
+; Vote cast in a Leios election
+leios_vote = persistent_vote / non_persistent_vote
+
+; Vote cast by a persistent voter (90 bytes)
+persistent_vote =
+  [ 0                        ; Vote type identifier for persistent voter
+  , election_id              ; 8-byte election identifier  
+  , endorser_block_hash      ; 32-byte hash of the endorser block being voted on
+  , persistent_voter_id      ; 2-byte epoch-specific pool identifier
+  , vote_signature           ; 48-byte BLS signature on (election_id || endorser_block_hash)
```

Comment:
The Rust and Haskell simulations use a different data structure called a "vote bundle" because they vote on many EBs in the pipeline. Instead of single `endorser_block_hash` and `vote_signature` entries, they have a `map` from `endorser_block_hash` to `vote_signature`.  This saves ~10 bytes per persistent vote. Do we want to adopt this in the spec, or not worry about the 10 bytes?

https://github.com/input-output-hk/ouroboros-leios/blob/b5dd5b4440b01d79960bf6de6573dcecfbce8d3a/sim-rs/sim-core/src/model.rs#L208-L212